tests/spark/test_mrjob_spark_harness.py

```diff
@@ -244,5 +267,54 @@ def test_passthru_args(self):
             input_bytes=input_bytes,
             job_args=['--chars', '--ignore', 'to'])
 
+    def test_combiner_called(self):
+        input_bytes = b'one two three one two three one two three'
+        job = self._harness_job(
+            MRWordFreqCountFailingCombiner, input_bytes=input_bytes)
+
+        with job.make_runner() as runner, \
+                self.assertRaises(Exception) as assert_raises_context:
+            runner.run()
+
+        exception_text = assert_raises_context.exception.__str__()
+        expected_str = MRWordFreqCountFailingCombiner.unique_exception_str
+        assert expected_str in exception_text
+
+    def test_combiner_yields_two_values(self):
+        input_bytes = b'one two three one two three one two three'
+
+        job = self._harness_job(MRWordFreqCountCombinerYieldsTwo,
+                                input_bytes=input_bytes)
+
+        # Given that the combiner for this job yields the count and 1000 for
+        # each word and the mrjob_spark_harness combiner_helper stops running
+        # jobs' combiners if they do not reduce the mapper's output to a single
+        # value per key, we expect the combiner to run once per word, resulting
+        # in an extra 1000 being added to each word's count.
+        #
+        # Note that if combiner_helper did not stop running the combiner in
+        # this case, we would expect 2000 to be added to each word's count
+        # since each word appears 3 times, resulting in 2 calls to
+        # combine_pairs.
+        with job.make_runner() as runner:
+            runner.run()
+
+            self.assertEqual(
+                dict(job.parse_output(runner.cat_output())),
+                dict(one=1003, two=1003, three=1003))
+
+    def test_combiner_yields_zero_values(self):
```

Comment:
Really wanted to see a test of adding an empty list to a single-item list. A combiner that summed integer values (and declined to spit out a key-value pair if the value was zero) would test this, *and* you could compare its output to that of running the job without the harness on the inline runner.

Reply:
This test is good too, wouldn't hurt to have both.